### ISSUE TITLE
Should file:// with hostname raise NotImplementedError?

### ISSUE THREAD

**Issue by timsnyder** [fsspec/filesystem_spec#998]
I was recently reminded that `file:` URIs can specify a hostname.  If a `file:` URI is passed with a hostname, it is currently dropped on the floor silently. 
 Should something `raise NotImplementedError` to avoid any confusion?  If so, I'm happy to create a test for this case and submit a PR.

**Comment by martindurant** [comment]
I suppose maybe? I don't have a strong feeling, I've never seen such URIs.

**Comment by timsnyder** [comment]
I agree that it is unlikely to encounter file: uri that specify a hostname.
I always forget that the RFC for file uri even allows for it. I was only
using this as an excuse to get familiar with the code.

https://datatracker.ietf.org/doc/html/rfc8089#section-3 is the section of
rfc-8089 that deals with file uri and local filesystem paths.

If I get a chance, I'll submit pr with a test and patch to for it just to
learn more about fsspec.

**Comment by martindurant** [comment]
👍
